dhall/tests/Dhall/Test/ETags.hs

```diff
@@ -0,0 +1,60 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Dhall.Test.ETags where
+
+import Data.Monoid ((<>))
+import Data.Text (Text)
+import Dhall.Util (Input(..))
+import Prelude hiding (FilePath)
+import Test.Tasty (TestTree)
+import Turtle (FilePath, format, fp)
+
+import qualified Data.Text        as Text
+import qualified Data.Text.IO     as Text.IO
+import qualified Dhall.ETags      as ETags
+import qualified Dhall.Test.Util  as Test.Util
+import qualified Test.Tasty       as Tasty
+import qualified Test.Tasty.HUnit as Tasty.HUnit
+import qualified Turtle
+
+etagsDirectory :: FilePath
+etagsDirectory = "./tests/etags"
+
+getTests :: IO TestTree
+getTests = do
+    etagsTests <- Test.Util.discover (Turtle.chars <* ".dhall") etagsTest (Turtle.lstree etagsDirectory)
+
+    let testTree = Tasty.testGroup "etags tests" [ etagsTests, etagsDirTest ]
+
+    return testTree
+
+etagsTest :: Text -> TestTree
+etagsTest prefix =
+    Tasty.HUnit.testCase (Text.unpack prefix) $ do
+        let inputFile  = Text.unpack (prefix <> ".dhall")
+        let outputFile = Text.unpack (prefix <> ".tags")
+
+        actualTags <- Test.Util.toDhallPath <$> ETags.generate (InputFile inputFile) [""] False
+
+        expectedTags <- readExpected outputFile
+
+        let message = "The actual tags did not match the expected tags"
+
+        Tasty.HUnit.assertEqual message expectedTags actualTags
+
+etagsDirTest :: TestTree
+etagsDirTest =
+    Tasty.HUnit.testCase "all" $ do
+        let outputFile = Text.unpack . format fp $ etagsDirectory Turtle.</> "all.tags"
+
+        actualTags <-  Test.Util.toDhallPath <$> ETags.generate (InputFile (Text.unpack . format fp $ etagsDirectory))
+                                                     [".dhall"] False
```

Comment:
This is more conventional and IMHO easier to read:

```haskell
        actualTags <-
            fmap Test.Util.toDhallPath
                (ETags.generate
                    (InputFile (Text.unpack . format fp $ etagsDirectory))
                    [".dhall"]
                    False)
```

Reply:
> Just for information, why `fmap` better than `<$>`? Or when `<$>` is more preferable?

I prefer `fmap` for multiline situations, mostly because I can't decide where to put the `<$>`.